### PR TITLE
Use setuptools to compile the simulator extension module

### PR DESCRIPTION
**Pull request by eric-wieser**
cc @themperek. This might be a dead end, but it shows a solution that might help simplify makefiles

### CODE REVIEW

File: cocotb/share/include/cocotb_utils.h [hunk 1]
```diff
@@ -54,7 +54,7 @@ void to_python(void) {
     //fprintf(stderr, "INFO: Calling up to python %d\n", is_python_context);
 }
 
-void to_simulator(void) {
+inline void to_simulator(void) {
```
Comment (eric-wieser) [inline cocotb/share/include/cocotb_utils.h]: This is #954

**Comment by themperek** [comment]
@eric-wieser Thanks.

Was thinking about this but there are things like this: https://github.com/potentialventures/cocotb/blob/9b68e181a37588dfed01690892c18420463dde64/cocotb/share/lib/fli/Makefile#L31
or
https://github.com/potentialventures/cocotb/blob/0fd76bbc0dfe7a4c4a4161810c30ed12e75ec2f5/cocotb/share/makefiles/simulators/Makefile.aldec#L157-L158

This would require to compile for every simulator separately at setup and assume all is installed at This can be done but ...

**Comment by eric-wieser** [comment]
@themperek: But those modules aren't in the dependency chain of `simulator`, right? AFAIK, there are two different phases of compilation:

* Compiling `simulator`, `libgpi`, `libgpilog`, and `libcocotbutils`
* Compiling test-benches, which uses the users RTL source and the simulator-specific shared libraries

It seems to me that we can do the first step possibly fully within setuptools, or at least compile the `lib*` libraries prior to running setuptools - that would allow the `simulator` module to be pip-installed.

**Comment by themperek** [comment]
`EXTRA_LIBS` and `EXTRA_LIBDIRS` are used for 'libvpi', `libvhpi` and `libfli`

I do not know how yet much this is need.

Maybe this is good enough to start? https://github.com/themperek/cocotb-test
Can debug all those dependencies (still will have requirement on gcc but this normally is not a big issue).

**Comment by eric-wieser** [comment]
> EXTRA_LIBS and EXTRA_LIBDIRS are used for libvpi, libvhpi and libfli

I think this is only true of `libfli`

**Comment by eric-wieser** [comment]
Closing in favor of #1306, which contains most of this.

All that's missing is the way that the `simulator` module is compiled, which with that patch is still not available without tweaking the path.

**Pull request closed without merging**